### PR TITLE
Remove ActiveIssue from tests.

### LINKED ISSUE #26820: Marshal.GetNativeVariantForObject throws ComException for various objects
No failures with `msbuild /T:BuildAndTest /P:TargetGroup=netfx`

But with `msbuild /T:BuildAndTest /P:TargetGroup=netcoreapp`
```  ===========================================================================================================
  xUnit.net console test runner (64-bit .NET Core)
  Copyright (C) 2014 Outercurve Foundation.

  Discovering: System.Runtime.InteropServices.Tests
  Discovered:  System.Runtime.InteropServices.Tests
  Starting:    System.Runtime.InteropServices.Tests
     System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests.GetNativeVariantForObject_RoundtrippingPrimiti
  ves_Success(primitive: [00:00:00.0000010], expectedVarType: 8228, expectedValue: -1) [FAIL]
        System.Runtime.InteropServices.COMException : Unspecified error (Exception from HRESULT: 0x80004005 (E_FAIL))
        Stack Trace:
              at System.Runtime.InteropServices.Marshal.GetNativeVariantForObject(Object obj, IntPtr pDstNativeVariant)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetNativeVariantForObjectTests.cs(194,0): at System.Runtime.InteropServices.Tests.GetNativeVariantForObjec
  tTests.GetNativeVariantForObject_ValidObject_Success(Object primitive, VarEnum expectedVarType, IntPtr expectedValue,
   Object expectedRoundtripValue)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetNativeVariantForObjectTests.cs(87,0): at System.Runtime.InteropServices.Tests.GetNativeVariantForObject
  Tests.GetNativeVariantForObject_RoundtrippingPrimitives_Success(Object primitive, VarEnum expectedVarType, IntPtr exp
  ectedValue)
     System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests.GetNativeVariantForObject_RoundtrippingPrimiti
  ves_Success(primitive: System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests+StructWithValue, expectedV
  arType: VT_RECORD, expectedValue: -1) [FAIL]
        System.Runtime.InteropServices.COMException : Unspecified error (Exception from HRESULT: 0x80004005 (E_FAIL))
        Stack Trace:
              at System.Runtime.InteropServices.Marshal.GetNativeVariantForObject(Object obj, IntPtr pDstNativeVariant)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetNativeVariantForObjectTests.cs(194,0): at System.Runtime.InteropServices.Tests.GetNativeVariantForObjec
  tTests.GetNativeVariantForObject_ValidObject_Success(Object primitive, VarEnum expectedVarType, IntPtr expectedValue,
   Object expectedRoundtripValue)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetNativeVariantForObjectTests.cs(87,0): at System.Runtime.InteropServices.Tests.GetNativeVariantForObject
  Tests.GetNativeVariantForObject_RoundtrippingPrimitives_Success(Object primitive, VarEnum expectedVarType, IntPtr exp
  ectedValue)
     System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests.GetNativeVariantForObject_RoundtrippingPrimiti
  ves_Success(primitive: [System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests+StructWithValue, null], e
  xpectedVarType: 8204, expectedValue: -1) [FAIL]
        System.Runtime.InteropServices.COMException : Unspecified error (Exception from HRESULT: 0x80004005 (E_FAIL))
        Stack Trace:
              at System.Runtime.InteropServices.Marshal.GetNativeVariantForObject(Object obj, IntPtr pDstNativeVariant)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetNativeVariantForObjectTests.cs(194,0): at System.Runtime.InteropServices.Tests.GetNativeVariantForObjec
  tTests.GetNativeVariantForObject_ValidObject_Success(Object primitive, VarEnum expectedVarType, IntPtr expectedValue,
   Object expectedRoundtripValue)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetNativeVariantForObjectTests.cs(87,0): at System.Runtime.InteropServices.Tests.GetNativeVariantForObject
  Tests.GetNativeVariantForObject_RoundtrippingPrimitives_Success(Object primitive, VarEnum expectedVarType, IntPtr exp
  ectedValue)
     System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests.GetNativeVariant_NotInteropCompatible_ThrowsAr
  gumentException(obj: [System.Runtime.InteropServices.Tests.Common.GenericStruct`1[System.String]]) [FAIL]
        Assert.Throws() Failure
        Expected: typeof(System.ArgumentException)
        Actual:   typeof(System.Runtime.InteropServices.COMException): Unspecified error (Exception from HRESULT: 0x800
  04005 (E_FAIL))
        Stack Trace:
              at System.Runtime.InteropServices.Marshal.GetNativeVariantForObject(Object obj, IntPtr pDstNativeVariant)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetNativeVariantForObjectTests.cs(343,0): at System.Runtime.InteropServices.Tests.GetNativeVariantForObjec
  tTests.<>c__DisplayClass21_0.<GetNativeVariant_NotInteropCompatible_ThrowsArgumentException>b__0()
     System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests.GetNativeVariant_NotInteropCompatible_ThrowsAr
  gumentException(obj: []) [FAIL]
        Assert.Throws() Failure
        Expected: typeof(System.ArgumentException)
        Actual:   typeof(System.Runtime.InteropServices.COMException): Unspecified error (Exception from HRESULT: 0x800
  04005 (E_FAIL))
        Stack Trace:
              at System.Runtime.InteropServices.Marshal.GetNativeVariantForObject(Object obj, IntPtr pDstNativeVariant)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetNativeVariantForObjectTests.cs(343,0): at System.Runtime.InteropServices.Tests.GetNativeVariantForObjec
  tTests.<>c__DisplayClass21_0.<GetNativeVariant_NotInteropCompatible_ThrowsArgumentException>b__0()
     System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests.GetNativeVariant_NotInteropCompatible_ThrowsAr
  gumentException(obj: [System.Runtime.InteropServices.Tests.Common.GenericStruct`1[System.String]]) [FAIL]
        Assert.Throws() Failure
        Expected: typeof(System.ArgumentException)
        Actual:   typeof(System.Runtime.InteropServices.COMException): Unspecified error (Exception from HRESULT: 0x800
  04005 (E_FAIL))
        Stack Trace:
              at System.Runtime.InteropServices.Marshal.GetNativeVariantForObject(Object obj, IntPtr pDstNativeVariant)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetNativeVariantForObjectTests.cs(343,0): at System.Runtime.InteropServices.Tests.GetNativeVariantForObjec
  tTests.<>c__DisplayClass21_0.<GetNativeVariant_NotInteropCompatible_ThrowsArgumentException>b__0()
     System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests.GetNativeVariant_NotInteropCompatible_ThrowsAr
  gumentException(obj: []) [FAIL]
        Assert.Throws() Failure
        Expected: typeof(System.ArgumentException)
        Actual:   typeof(System.Runtime.InteropServices.COMException): Unspecified error (Exception from HRESULT: 0x800
  04005 (E_FAIL))
        Stack Trace:
              at System.Runtime.InteropServices.Marshal.GetNativeVariantForObject(Object obj, IntPtr pDstNativeVariant)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetNativeVariantForObjectTests.cs(343,0): at System.Runtime.InteropServices.Tests.GetNativeVariantForObjec
  tTests.<>c__DisplayClass21_0.<GetNativeVariant_NotInteropCompatible_ThrowsArgumentException>b__0()
     System.Runtime.InteropServices.Tests.GetNativeVariantForObjectTests.GetNativeVariant_NotInteropCompatible_ThrowsAr
  gumentException(obj: [Color [A=0, R=0, G=0, B=10]]) [FAIL]
        Assert.Throws() Failure
        Expected: typeof(System.ArgumentException)
        Actual:   typeof(System.Runtime.InteropServices.COMException): Unspecified error (Exception from HRESULT: 0x800
  04005 (E_FAIL))
        Stack Trace:
              at System.Runtime.InteropServices.Marshal.GetNativeVariantForObject(Object obj, IntPtr pDstNativeVariant)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetNativeVariantForObjectTests.cs(343,0): at System.Runtime.InteropServices.Tests.GetNativeVariantForObjec
  tTests.<>c__DisplayClass21_0.<GetNativeVariant_NotInteropCompatible_ThrowsArgumentException>b__0()
  Finished:    System.Runtime.InteropServices.Tests

```

### LINKED ISSUE #27015: Marshal.GetObjectForNativeVariant with IRecordInfo doesn't match netfx behaviour
From dotnet/corefx#31461

```cs
[Fact]
[PlatformSpecific(TestPlatforms.Windows)]
[ActiveIssue()]
public void GetObjectForNativeVariant_Record_ReturnsExpected()
{
    int record = 10;
    var recordInfo = new RecordInfo { Guid = typeof(int).GUID };
    IntPtr pRecord = Marshal.AllocHGlobal(Marshal.SizeOf<int>());
    IntPtr pRecordInfo = Marshal.GetComInterfaceForObject<RecordInfo, IRecordInfo>(recordInfo);
    try
    {
        Marshal.StructureToPtr(record, pRecord, fDeleteOld: false);

        Variant variant = CreateVariant(VT_RECORD, new UnionTypes
        {
            _record = new Record
            {
                _record = pRecord,
                _recordInfo = pRecordInfo
            }
        });
        Assert.Equal(10, GetObjectForNativeVariant(variant));
        GetObjectForNativeVariant_NestedVariant_ReturnsExpected(variant, record);

        variant.m_Variant.vt |= VT_BYREF;
        Assert.Equal(10, GetObjectForNativeVariant(variant));
    }
    finally
    {
        Marshal.DestroyStructure<int>(pRecord);
        Marshal.FreeHGlobal(pRecord);
        Marshal.Release(pRecordInfo);
    }
}
```

This passes with netfx

On netcoreapp:
```
Discovering: System.Runtime.InteropServices.Tests
  Discovered:  System.Runtime.InteropServices.Tests
  Starting:    System.Runtime.InteropServices.Tests
     System.Runtime.InteropServices.Tests.GetObjectForNativeVariantTests.GetObjectForNativeVariant_Record_ReturnsExpect
  ed [FAIL]
        System.ArgumentException : The specified record cannot be mapped to a managed value class.
        Stack Trace:
              at System.Runtime.InteropServices.Marshal.GetObjectForNativeVariant(IntPtr pSrcNativeVariant)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetObjectForNativeVariantTests.cs(1027,0): at System.Runtime.InteropServices.Tests.GetObjectForNativeVaria
  ntTests.GetObjectForNativeVariant(Variant variant)
           C:\Users\hughb\Documents\GitHub\corefx\src\System.Runtime.InteropServices\tests\System\Runtime\InteropServic
  es\Marshal\GetObjectForNativeVariantTests.cs(700,0): at System.Runtime.InteropServices.Tests.GetObjectForNativeVarian
  tTests.GetObjectForNativeVariant_Record_ReturnsExpected()
  Finished:    System.Runtime.InteropServices.Tests
```

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Update tests to pass since issues are being marked "Won't Fix".

Close https://github.com/dotnet/runtime/issues/26820
Close https://github.com/dotnet/runtime/issues/27015

/cc @elinor-fung @jkoritzinsky

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]

File: src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetObjectForNativeVariantTests.Windows.cs [hunk 1]
```diff
@@ -538,7 +538,7 @@ public void GetObjectForNativeVariant_NestedVariant_ReturnsExpected(Variant sour
         }
 
         [Fact]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/27015", TargetFrameworkMonikers.Netcoreapp)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Netcoreapp, "VT_RECORD not supported on .NET Core")]
```
Comment (jkotas) [inline src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetObjectForNativeVariantTests.Windows.cs:541]: These tests run on .NET Core only. If they are not relevant for .NET Core, delete them (or change them to check for behavior expected on .NET Core).

Comment (AaronRobinsonMSFT) [reply src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetObjectForNativeVariantTests.Windows.cs:541]: This code path cannot be run on .NET Core due to the reasons mentioned in https://github.com/dotnet/runtime/issues/27015#issuecomment-611079237. Will remove this specific test.

Comment (AaronRobinsonMSFT) [reply src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetObjectForNativeVariantTests.Windows.cs]: Instead I changed the test to validate the failure.

File: src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs [hunk 2]
```diff
@@ -119,10 +119,6 @@ public static IEnumerable<object[]> GetNativeVariantForObject_NonRoundtrippingPr
             {
```
Comment (jkotas) [inline src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs:119]: These `!PlatformDetection.IsNetCore` checks can be delete too.

Comment (jkotas) [reply src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs:119]: And tests moved to the methods later in the file that validate the failure?

Comment (AaronRobinsonMSFT) [reply src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs]: > And tests moved to the methods later in the file that validate the failure?

I don't quite get what this means.

File: src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs [hunk 3]
```diff
@@ -183,7 +163,6 @@ public static IEnumerable<object[]> GetNativeVariantForObject_NonRoundtrippingPr
         [Theory]
         [MemberData(nameof(GetNativeVariantForObject_NonRoundtrippingPrimitives_TestData))]
         [PlatformSpecific(TestPlatforms.Windows)]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/26820")]
         public void GetNativeVariantForObject_ValidObject_Success(object primitive, VarEnum expectedVarType, IntPtr expectedValue, object expectedRoundtripValue)
         {
             var v = new Variant();
```
Comment (jkotas) [inline src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs:168]: `GetNativeVariant_NotInteropCompatible_TestData` method in this file is dead code.

Comment (AaronRobinsonMSFT) [reply src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs:168]: Actually so is `GetNativeVariantForObject_RoundtrippingPrimitives_TestData`.

File: src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs (hunk 2 above)
Comment (jkotas) [reply src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs]: I meant that we should preserve the test cases being removed, but change the expected behavior for them.

I assume that the expected behavior for these test cases is some kind of exception. There are theories later in this file that are testing different types of exceptions, so we may be able to just move the the cases to there theories.

(Ignore this if we have the appropriate coverage for these cases already somewhere.0

File: src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs [hunk 4]
```diff
@@ -15,71 +15,6 @@ namespace System.Runtime.InteropServices.Tests
...
-            yield return new object[] { valueType, VarEnum.VT_RECORD, (IntPtr)(-1) };
-
-            var genericClass = new GenericClass<string>();
-            yield return new object[] { new object[] { nonGenericClass, genericClass, null }, (VarEnum)8204, (IntPtr)(-1) };
-
-            yield return new object[] { new object[] { valueType, null }, (VarEnum)8204, (IntPtr)(-1) };
-
-            // Delegate.
-            MethodInfo method = typeof(GetNativeVariantForObjectTests).GetMethod(nameof(NonGenericMethod), BindingFlags.NonPublic | BindingFlags.Static);
-            Delegate d = method.CreateDelegate(typeof(NonGenericDelegate));
```
Comment (elinor-fung) [inline src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs:79]: It looks like `StructWithValue`, `NonGenericMethod`, and `NonGenericDelegate` can be removed.

File: src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs [hunk 5]
```diff
@@ -115,28 +50,12 @@ public static IEnumerable<object[]> GetNativeVariantForObject_NonRoundtrippingPr
...
             yield return new object[] { new BStrWrapper(null), VarEnum.VT_BSTR, IntPtr.Zero, null };
 
             yield return new object[] { new UnknownWrapper[] { new UnknownWrapper(null), new UnknownWrapper(10) }, (VarEnum)8205, (IntPtr)(-1), new object[] { null, 10 }  };
-            if (!PlatformDetection.IsNetCore)
-            {
-                yield return new object[] { new DispatchWrapper[] { new DispatchWrapper(null), new DispatchWrapper(10) }, (VarEnum)8201, (IntPtr)(-1), new object[] { null, 10 } };
-            }
-            else
-            {
-                Assert.Throws<PlatformNotSupportedException>(() => new DispatchWrapper(10));
```
Comment (elinor-fung) [inline src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs:138]: Creating a `DispatchWrapper` should no longer throw `PlatformNotSupportedException` on Windows in .NET Core, so I think the two cases currently under `!PlatformDetectior.IsNetCore` should be valid for Windows now?

Comment (elinor-fung) [reply src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs:138]: And it looks like the only test that uses this is scoped to Windows.

**Review by elinor-fung — APPROVED** [review]

File: src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs (hunk 5 above)
Comment (AaronRobinsonMSFT) [reply src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs:138]: It is valid, but not for `Int32`. I would need to find some type that supports IDispatch first.

Comment (elinor-fung) [reply src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs:138]: `IDispatchComObject`?

Comment (elinor-fung) [reply src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetNativeVariantForObjectTests.cs:138]: Oh, wait, it looks there is `GetNativeVariantForObject_WrappedComObject_Success` for Windows.

**Review by stephentoub — APPROVED** [review]

**Comment by AaronRobinsonMSFT** [comment]
The formatting job was not impacted by this change.

**Pull request merged**